webservices/react-testing.md

```diff
@@ -0,0 +1,431 @@
+import IconFlag from "components/UI/IconFlag";
+import Portion from "components/UI/Portion";
+
+describe("MealEvent", () => {
+  let props, wrapper;
+
+  beforeEach(() => {
+    props = {
+      dateTimeText: "9:37 AM",
+      flagged: true,
+      mealEventImgOnError: jest.fn(),
+      noteText: "I am a note",
+      onClick: jest.fn(),
+      portion: "Regular",
+      srcMealEvent: "http://via.placeholder.com/350x150",
+      stylesImage: "imageMealEvent",
+      stylesNoteText: "noteTextAbsent",
+      testMeal: false,
+      timeText: "Log Time: 10:39 AM on Jun 1, 2017"
+    };
+    wrapper = shallow(<MealEvent {...props} />);
+  });
+
+  it("renders dateTimeText", () => {
+    expect(wrapper).toIncludeText(props.dateTimeText);
+  });
+
+  it("renders noteText", () => {
+    expect(wrapper).toIncludeText(props.noteText);
+  });
+
+  it("renders timeText", () => {
+    expect(wrapper).toIncludeText(props.timeText);
+  });
+
+  it("handles click events", () => {
+    wrapper.simulate("click");
+
+    expect(props.onClick).toHaveBeenCalled();
+  });
+
+  it("renders a span with className of stylesNoteText", () => {
+    const element = wrapper
+      .find("span")
+      .filterWhere(n => n.props().className === props.stylesNoteText);
+
+    expect(element).toBePresent();
+  });
+
+  it("renders img whose src is srcMealEvent", () => {
+    const element = wrapper.find("img").filterWhere(n => {
+      const { alt, className, src, onError } = n.props();
+      return (
+        className === props.stylesImage &&
+        src === props.srcMealEvent &&
+        onError === props.mealEventImgOnError &&
+        alt === text.altMeal
+      );
+    });
+
+    expect(element).toBePresent();
+
+    element.simulate("error");
+
+    expect(props.mealEventImgOnError).toHaveBeenCalled();
+  });
+
+  it("does not render an img whose src is srcPortion if testMeal is true", () => {
+    wrapper.setProps({ ...props, testMeal: true });
+    const element = wrapper
+      .find("img")
+      .filterWhere(n => n.props().src === props.srcPortion);
+
+    expect(element).not.toBePresent();
+  });
+
+  it("renders IconFlag if flagged prop is true", () => {
+    const element = wrapper.find(IconFlag);
+
+    expect(element).toBePresent();
+  });
+
+  it("does not render IconFlag if flagged prop is false", () => {
+    wrapper.setProps({ ...props, flagged: false });
+    const element = wrapper.find(IconFlag);
+
+    expect(element).not.toBePresent();
+  });
+
+  it("renders Portion if test meal false", () => {
+    const element = wrapper.find(Portion);
+
+    expect(element).toBePresent();
+
+    const { portion } = element.props();
+
+    expect(portion).toBe(props.portion);
+  });
+
+  it("does not render Portion if test meal true", () => {
+    wrapper.setProps({ ...props, testMeal: true });
+    const element = wrapper.find(Portion);
+
+    expect(element).not.toBePresent();
+  });
+});
+```
+
+### Testing enhance HOC
```

Comment:
what's HOC stand for?

Reply:
[Higher-Order component](https://reactjs.org/docs/higher-order-components.html). I'll spell it out here, but wont go into much detail because @hchood is writing the recompose section which heavily delves into that